sway-core/src/ir_generation/function.rs

```diff
@@ -2774,59 +2720,44 @@ impl<'eng> FnCompiler<'eng> {
         ));
 
         // Local pointer to hold the array of b256s
-        let value_ptr = self
+        let local_var = self
             .function
-            .new_local_ptr(context, alias_value_name, b256_array_type, true, None)
+            .new_local_var(context, local_value_name, b256_array_type, None)
             .map_err(|ir_error| CompileError::InternalOwned(ir_error.to_string(), Span::dummy()))?;
 
-        // Convert the local pointer created to a value of the original type using
-        // get_ptr.
-        let value_ptr_val = self
+        // Convert the local pointer created to a value of the original type using cast_ptr.
+        let local_val = self
             .current_block
             .ins(context)
-            .get_ptr(value_ptr, *r#type, 0)
+            .get_local(local_var)
+            .add_metadatum(context, span_md_idx);
+        let final_val = self
+            .current_block
+            .ins(context)
+            .cast_ptr(local_val, *r#type, 0)
             .add_metadatum(context, span_md_idx);
 
-        for array_index in 0..number_of_elements {
-            if array_index > 0 {
-                // Prepare key for the next iteration but not for array index 0
-                // because the first key was generated earlier.
-                // Const value for the key from the initial hash + array_index
-                let const_key = convert_literal_to_value(
-                    context,
-                    &Literal::B256(*add_to_b256(*storage_key, array_index)),
-                )
-                .add_metadatum(context, span_md_idx);
-
-                // Convert the key pointer to a value using get_ptr
-                let key_ptr_ty = *key_ptr.get_type(context);
-                *key_ptr_val = self
-                    .current_block
-                    .ins(context)
-                    .get_ptr(*key_ptr, key_ptr_ty, 0)
-                    .add_metadatum(context, span_md_idx);
-
-                // Store the const hash value to the key pointer value
-                self.current_block
-                    .ins(context)
-                    .store(*key_ptr_val, const_key)
-                    .add_metadatum(context, span_md_idx);
-            }
-
+        if number_of_elements > 0 {
             // Get the b256 from the array at index iter
-            let value_ptr_val_b256 = self
+            let value_val_b256 = self
                 .current_block
                 .ins(context)
-                .get_ptr(value_ptr, Type::B256, array_index)
```

Comment:
Yeah I remember one of the main reasons for adding `get_ptr` was this particular use case.